sechub-api-java/src/main/java/com/mercedesbenz/sechub/api/java/demo/cli/SecHubClientConfigurationFactory.java

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+package com.mercedesbenz.sechub.api.java.demo.cli;
+
+import com.mercedesbenz.sechub.api.java.SecHubClient;
+import com.mercedesbenz.sechub.api.java.demo.cli.exceptions.SecHubClientConfigurationRuntimeException;
+
+public class SecHubClientConfigurationFactory {
+
+    EnvironmentVariableReader environmentVariableReader;
+
+    public SecHubClientConfigurationFactory() {
+        environmentVariableReader = new EnvironmentVariableReader();
+    }
+
+    public SecHubClient create(CommandLineSettings settings) {
+        String serverUri = settings.getServerUri();
+        int serverPort = settings.getServerPort();
+        String userId = settings.getUserId();
+        String apiToken = settings.getApiToken();
+
+        if (serverUri == null) {
+            serverUri = environmentVariableReader.readAsString(EnvironmentVariableConstants.SECHUB_SERVER_URI);
```

Comment:
Java does have an URI type which you could parse in addition to ensure it is well formatted. Furthermore, it might make sense to ignore some part of the URI (for example, query parameters) and check the protocol (needs to be HTTP(S)).

For example by using `URI.create()`: https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#create-java.lang.String-